packages/grpc-native-core/src/common.js

```diff
@@ -112,6 +112,35 @@ exports.getMethodType = function(method_definition) {
   }
 };
 
+const PEM_CERT_HEADER = "-----BEGIN CERTIFICATE-----";
+const PEM_CERT_FOOTER = "-----END CERTIFICATE-----";
+
+/**
+ * Converts a certificate in ASCII PEM format to a byte buffer.
+ * @param {string} cert The certificate in PEM format
+ * @return {Buffer|Error} The certificate as a byte buffer.
+ */
+
+exports.pemCertificateToRaw = function(cert) {
+  // Parse cert from pem to a version that matches the tls.checkServerIdentity
+  // format and the tls.TLSSocket.getPeerCertificate format.
+  // https://nodejs.org/api/tls.html#tls_tls_checkserveridentity_hostname_cert
+  // https://nodejs.org/api/tls.html#tls_tlssocket_getpeercertificate_detailed
+
+  var pemHeaderIndex = cert.indexOf(PEM_CERT_HEADER);
+  if (pemHeaderIndex === -1) {
+    return new Error("Unable to parse certificate PEM.");
+  }
+  cert = cert.substring(pemHeaderIndex);
+  var pemFooterIndex = cert.indexOf(PEM_CERT_FOOTER);
+  if (pemFooterIndex === -1) {
+    return new Error("Unable to parse certificate PEM.");
```

Comment:
Why return an error here, instead of just throwing?

Reply:
This is not new code :-) You're reviewing me moving code from #403, so you're arguing about code written by @JackOfMostTrades here :-D

Reply:
This doesn't even do the same thing as the code you copied, because adding another layer of functions changes what `return` means.

Reply:
Pff, you're correct. My mistake. Let me fix this.